src/main/kotlin/com/autonomousapps/tasks/ComputeAdviceTask.kt

```diff
@@ -254,10 +262,79 @@ internal class DependencyAdviceBuilder(
 
   private fun computeDependencyAdvice(projectPath: String): Sequence<Advice> {
     val declarations = declarations.filterToSet { Configurations.isForRegularDependency(it.configurationName) }
-    return dependencyUsages.asSequence()
+
+    /*
+     * To support KMP properly, we need to do an initial pass at deps to clean up any KMP deps advice.
+     *
+     * The primary case we are trying to catch and avoid here is one where we "fix" canonical KMP deps by trying to replace
+     * them with added deps on their specific targets ("-jvm", "-android", etc). To solve this, we save off information
+     * about both in the below mappings and then cross-reference them in the second pass farther down.
+     *
+     * Finally, in the third pass, we do our usual processing of advice but with the cleaned KMP advice. It's
+     * important we clean up the KMP advice first so that the usual processing can continue to account for bundles
+     * in a simple way.
+     */
+
+    // Deferred KMP advice.
+    val deferredKmpAdvice = mutableListOf<Pair<Advice, Coordinates>>()
```

Comment:
- [ ] TODO for me: open this up in an IDE and make sure I understand the changes 😅